content/docs/intro/concepts/config.md

```diff
@@ -430,3 +430,61 @@ For example, this configures a stack to use a HashiCorp Vault transit key named
+
+First, you will need to get copies of the two assets above with plaintext copies of your secret values from your current stack.  This is a sensitive operation, and you should be careful to do this in a safe environment and be sure to remove these files when you are done.
+
+```
+$ pulumi config --show-secrets --json > config.json
+$ pulumi stack export --show-secrets > stack.json
+```
+
+Next, you will need to create a temporary new stack using the desired new secrets provider.  If using the `passphrase` secrets provider, this can also be used to create a new passphrase to use for encryption.
+
+```
+$ pulumi stack init temp --secrets-provider <provider>
+```
+
+If you chose `passphrase` you will see something like this:
+
+```
+$ cat Pulumi.temp.yaml
+encryptionsalt: v1:h8kauiCOXZc=:v1:0Tjy4uHvwgw/b0Hp:gKAqRJ74vyIkc5BeEUlCFXynwRTPFQ==
+```
+
+Else if you chose one of the cloud-provider secrets providers you will see something like this:
+
+```
+$ cat Pulumi.temp.yaml
+secretsprovider: awskms://alias/my-kms-key
+encryptedkey: AQICAHhivZtfYgNwNJDVWDd++88uo2qgy5MJqqzsPfTFv7uzDgF+1kaw9/wSKgGh5QpT/cwtAAAAfjB8BgkqhkiG9w0BBwagbzBtAgEAMGgGCSqGSIb3DQEHATAeBglghkgBZQMEAS4wEQQMc2EiH7Mli5ohzB1aAgEQgDsOszkuuC7wffN3FdtKCNAcRoJq3krWmeA/ucnSMn/F2OYVIDKxDZkjJYZBAjEmpWL49Fcly/n8NMNHhw==
+```
+
+And if you chose the Pulumi Service secrets provider, there will be no stack file.
+
+You will use the values above to replace those in use in your existing stack.  
+
+First, your config.  
+
+1. You will need to remove all encypted secrets from your current config using `pulumi config rm` or by removing them by hand from the `.yaml` file (you saved them to `config.json` above).  
+2. Then copy the lines above (`encyptionsalt` or `secretsprovider`+`encryptedkey`) from the `temp` stack into your primary stack (replacing the existing corresponding lines). 
+3. Save that, and then run `pulumi config set --secret <key> <value>` for each of the secret config values you removed, using the values from `config.json`.  
+
+Next, your state.  
+
+1. Open `stack.json` and find the `secrets_providers` section.  
+2. Based on the output of your `Pulumi.temp.yaml` pick from these options:
+    * For `encyrptionsalt` set `"type": "passphrase",` and replace the contents of the `state` with `"salt": "<value>"`.
+    * For `secretsprovider`+`encryptedkey` set `"type": "cloud"` and replace the contents of the `state` with `"url": "<value>"` and `"encryptedkey": "<value>"` respectively.
+    * For Pulumi Service secrets-provider, set `"type": "service"` and replace the contents of the `state` with `"url"`, `"owner"`, `"project"` and `"stack"` as appropriate for your existing stack.
+
+Save your `stack.json` with those changes, then run `pulumi stack import < stack.json` to import it into your Pulumi stack. 
```

Comment:
The `pulumi stack init ...` before getting to this point will change the actively selected stack, so I think we need to do a `pulumi stack select` prior to the `import` to go back to the "real" stack, right?